File: sdk/typescript/src/resources/paymentLinks.ts
import type { MashgateClient } from "../client.js";
import type { PaymentLink } from "../types.js";

export interface CreatePaymentLinkRequest {
  tenantId: string;
  amount: number;
  currency: string;
  description?: string;
  expiresAt?: string;
}

export class PaymentLinksResource {
  constructor(private readonly client: MashgateClient) {}

  async create(data: CreatePaymentLinkRequest): Promise<PaymentLink> {
    return this.client.request<PaymentLink>("POST", "/v1/payment-links", { body: data });
  }

  async list(tenantId: string): Promise<PaymentLink[]> {
    return this.client.request<PaymentLink[]>("GET", "/v1/payment-links", { query: { tenantId } });
  }

  async get(id: string): Promise<PaymentLink> {
    return this.client.request<PaymentLink>("GET", `/v1/payment-links/${id}`);
  }
}

File: sdk/python/mashgate/resources/payment_links.py
"""Payment links resource.

Shareable payment URLs, mirroring the payment-link endpoints on the
checkout-service exposed via the gateway.
"""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from mashgate.client import MashgateClient


class PaymentLinksResource:
    def __init__(self, client: MashgateClient) -> None:
        self._c = client

    def create(
        self,
        *,
        tenant_id: str,
        amount: int,
        currency: str,
        description: str | None = None,
        expires_at: str | None = None,
    ) -> dict[str, Any]:
        """Create a new payment link."""
        body: dict[str, Any] = {
            "tenantId": tenant_id,
            "amount": amount,
            "currency": currency,
        }
        if description is not None:
            body["description"] = description
        if expires_at is not None:
            body["expiresAt"] = expires_at
        return self._c.request("POST", "/v1/payment-links", body=body)

    def list(self, tenant_id: str) -> dict[str, Any]:
        """Return all payment links for a tenant."""
        return self._c.request("GET", "/v1/payment-links", query={"tenantId": tenant_id})

    def get(self, link_id: str) -> dict[str, Any]:
        """Return a payment link by ID."""
        return self._c.request("GET", f"/v1/payment-links/{link_id}")
